cmd/apply.go

```diff
@@ -42,41 +42,42 @@ func ApplyCmd() *cobra.Command {
 	This command will attempt to find an API model in a defined state store, and then apply any changes needed directly to a cloud.
 	The apply will run once, and ultimately time out if something goes wrong.`,
 		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) == 0 {
-				ao.Name = cli.StrEnvDef("KUBICORN_NAME", "")
-			} else if len(args) > 1 {
+			switch len(args) {
+			case 0:
+				ao.Name = viper.GetString(keyKubicornName)
+			case 1:
+				ao.Name = args[0]
+			default:
 				logger.Critical("Too many arguments.")
 				os.Exit(1)
-			} else {
-				ao.Name = args[0]
 			}
 
-			err := RunApply(ao)
-			if err != nil {
+			if err := runApply(ao); err != nil {
 				logger.Critical(err.Error())
 				os.Exit(1)
 			}
-
 		},
 	}
 
-	applyCmd.Flags().StringVarP(&ao.StateStore, "state-store", "s", cli.StrEnvDef("KUBICORN_STATE_STORE", "fs"), "The state store type to use for the cluster")
-	applyCmd.Flags().StringVarP(&ao.StateStorePath, "state-store-path", "S", cli.StrEnvDef("KUBICORN_STATE_STORE_PATH", "./_state"), "The state store path to use")
-	applyCmd.Flags().StringVarP(&ao.Set, "set", "e", cli.StrEnvDef("KUBICORN_SET", ""), "set cluster setting")
-	applyCmd.Flags().StringVar(&ao.AwsProfile, "aws-profile", cli.StrEnvDef("AWS_PROFILE", ""), "The profile to be used as defined in $HOME/.aws/credentials")
-	applyCmd.Flags().StringVar(&ao.GitRemote, "git-config", cli.StrEnvDef("KUBICORN_GIT_CONFIG", "git"), "The git remote url to be used for saving the git state for the cluster.")
+	fs := applyCmd.Flags()
```

Comment:
I'm wondering about a thing that could be addressed in another PR.

Could we move those common definitions to some function, and then, to invoke that function from here?
Those variables are re-used in almost every file, so maybe it would make sense if that's possible.

It's good to merge PR as it is, but this could be a good thing to work on if you're interested.